neuralphys.f90

```diff
@@ -0,0 +1,318 @@
+          nn_input_vector(521)    = sin(2.0* pi * doy/365.0)
+          nn_input_vector(522)    = cos(2.0* pi * hour/24.0)
+          nn_input_vector(523)    = cos(2.0* pi * doy/365.0)
+
+          nn_output_vector(:)     = 0.0
+
+          !print*,'NN corr', maxval(nn_input_vector(1:127)), minval(nn_input_vector(1:127)), nn_input_vector(128), maxval(nn_input_vector(129:255)), minval(nn_input_vector(129:255)), maxval(nn_input_vector(256:382)), minval(nn_input_vector(256:382)), maxval(nn_input_vector(383:509)), minval(nn_input_vector(383:509)), nn_input_vector(510:523)
+          !print*, 'NN corr in ', nn_input_vector(510:523)
+!             
+! Call NN computation
+          call compute_nn(nn_input_vector,nn_output_vector) !,nn_num_of_members,& 
+!
+! Unpack NN output vector
+          !nn_output_vector(:) = 0.
+          gu0(:)       = ugrs(:) + nn_output_vector(128:254)*dt ! u component of layer wind
+          gv0(:)       = vgrs(:) + nn_output_vector(255:381)*dt ! v component of layer wind
+          gt0(:)       = tgrs(:) + nn_output_vector(1:127)*dt   ! layer mean temperature 
+          oshm(:)      = shm(:)  + nn_output_vector(382:508)*dt ! specific humidity
+!
+          !print*,'NN corr out', maxval(nn_output_vector(1:127)), minval(nn_output_vector(1:127))
+          !print*,'NN corr out', maxval(nn_output_vector(128:254)), minval(nn_output_vector(128:254))
+          !print*,'NN corr out', maxval(nn_output_vector(255:381)), minval(nn_output_vector(255:381))
+          !print*,'NN corr out', maxval(nn_output_vector(382:508)), minval(nn_output_vector(382:508))
+          !if (me == 0)  print*,'Module NEURALPHYS: Before end eval_nn'
+        end subroutine eval_nn
+       
+        subroutine relu(x,y)
+          real(kind=4), intent(in) :: x
+          real(kind=4), intent(out) :: y
+          y = max(0.,x)
+        end subroutine relu
+
+        subroutine relum(x,y)
+          integer i
+          real(kind=4), intent(in) :: x(4096)
+          real(kind=4), intent(out) :: y(4096)
+          do i =1,4096
+             y(i) = max(x(i),0.0)
+          end do
+        end subroutine relum
+
+ 
+        subroutine  compute_nn(X,Y) !,num_of_members,w1,w2,b1,b2,nhid)
+ !  Input:
+ !            X(IN) NN input vector 
+          real(kind=4), intent(in):: X(523)
+          !integer, intent(in):: me
+
+ !   Ouput:
+ !            Y(OUT) NN output vector (composition coefficients for SNN)
+
+          real(kind=4), intent(out):: Y(508)
+
+! Local variables 
+          integer i, nout
+          real(kind=4) :: x_tmp1(4096), x_tmp2(4096)! x2(:),x3(:)
+          !real(kind=kind_phys), allocatable :: x_tmp1(:), x_tmp2(:)! x2(:),x3(:)
+          integer member,layer, l0, l1
+          !print*,"NEURAL: size:", size(X), size(Y)
+          !print*,"NN:", maxval(X(1:127)), X(128), maxval(X(129:255)),maxval(X(256:382)), maxval(X(383:509)), X(510:523)
+          Y(:) = 0.0
+
+          do member = 1,4 !nn_num_of_members
```

Comment:
i wonder why are we looping through 4 input vectors?

Reply:
This is looping through 4 nns for each prognostic variable. The name "member" is indeed confusing. will change to something else.